pkg/ocm/pullsecret_auth.go

```diff
@@ -0,0 +1,85 @@
+package ocm
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/patrickmn/go-cache"
+	"github.com/sirupsen/logrus"
+)
+
+type OCMAuthentication interface {
+	AuthenticatePullSecret(ctx context.Context, pullSecret string) (user *AuthPayload, err error)
+}
+
+type authentication struct {
+	client *Client
+}
+
+func (a authentication) AuthenticatePullSecret(ctx context.Context, pullSecret string) (user *AuthPayload, err error) {
+
+	authUser, found := a.client.Cache.Get(pullSecret)
+	if found {
+		return authUser.(*AuthPayload), nil
+	}
+	con := a.client.connection
```

Comment:
I think this is problematic.
This connection might be disconnected. Who will re-connect?
Whenever OCM is restarted it seems like we will not recover

Maybe it's better to create the connection for every request?

Reply:
We might have connections leak here.
We are creating new connections without closing the old ones.
IMHO the proper way to do this is to return the connection from NewConnection() and and not use use a client level connection. Then we can make sure we close this local connection instance using a `defer`